ui/src/app/pages/artifactVersion/components/modals/generateClientModal.tsx

```diff
@@ -0,0 +1,293 @@
+                                    onChange={this.onNamespaceChange}
+                                />
+                            </FormGroup>
+                        </GridItem>
+
+                        <FormGroup
+                            label="Language"
+                            fieldId="form-language"
+                            isRequired={true}
+                        >
+                            <div>
+                                <Dropdown
+                                    toggle={
+                                        <DropdownToggle id="form-type-toggle" data-testid="form-type-toggle" onToggle={this.onLanguageToggle} toggleIndicator={CaretDownIcon}>
+                                            { this.state.data.language ? this.state.data.language : "Java" }
+                                        </DropdownToggle>
+                                    }
+                                    onSelect={this.onLanguageSelect}
+                                    isOpen={this.state.languageIsExpanded}
+                                    dropdownItems={[
+                                        <DropdownItem id="Java" key="Java" data-testid="form-type-auto"><i>Java</i></DropdownItem>,
+                                        <DropdownItem id="CSharp" key="CSharp" data-testid="form-type-auto"><i>CSharp</i></DropdownItem>,
+                                        <DropdownItem id="Go" key="Go" data-testid="form-type-auto"><i>Go</i></DropdownItem>,
+                                        <DropdownItem id="Python" key="Python" data-testid="form-type-auto"><i>Python</i></DropdownItem>,
+                                        <DropdownItem id="Ruby" key="Ruby" data-testid="form-type-auto"><i>Ruby</i></DropdownItem>,
+                                        <DropdownItem id="TypeScript" key="TypeScript" data-testid="form-type-auto"><i>TypeScript</i></DropdownItem>,
+                                        <DropdownItem id="PHP" key="PHP" data-testid="form-type-auto"><i>PHP</i></DropdownItem>,
+                                        <DropdownItem id="Swift" key="Swift" data-testid="form-type-auto"><i>Swift</i></DropdownItem>,
+                                    ]}
+                                />
+                            </div>
+                        </FormGroup>
+
+                    </Grid>
+                </Form>
+            </Modal>
+        );
+    }
+
+    protected initializeState(): GenerateClientModalState {
+        return {
+            data: {
+                content: this.props.artifactContent,
+                clientClassName: "MyClient",
+                namespaceName: "io.dummy",
+                language: "Java"
+            },
+            languageIsExpanded: false,
+            isValid: true,
+            downloadLink: "",
+            isErrorVisible: false,
+            isDownloadLinkVisible: false,
+        };
+    }
+
+    private doGenerate = async (): Promise<void> => {
```

Comment:
from a UX perspective, having some kind of indication something is happening in the background would be helpful I think.

Reply:
agree, added a spinner 👍